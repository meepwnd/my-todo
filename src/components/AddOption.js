import React from 'react';
import {addOption} from '../actions/actions';
import {connect} from 'react-redux';
import uuid from 'uuid';

class AddOption extends React.Component {
	state = {
		error: ''
	}

	onSubmit = (e) => {
		e.preventDefault();
		const option = e.target.elements.inputOption.value.trim();
		const id = uuid();
		if(!option) {
 			this.setState(() => ({
 				error: 'Enter valid option!'
 			})) 
 		} else if (this.props.options.includes(option)) {
 			this.setState(() => ({
 				error: 'This option already exists!'
 			})) 
 		} else {
 			this.props.addOption(option, id);
 			e.target.elements.inputOption.value = '';
 			this.setState(() => ({
 				error: ''
 			})) 
 		}	
	}
	
	render(){
		return (
			<form onSubmit={this.onSubmit}>
			{this.state.error && <p className="error">{this.state.error}</p>}
				<input className="option-input" type='text' name='inputOption'/>
				<button className="btn-add-option">add option</button>
			</form>
		)
	}
}

const mapStateToPops = (state) => ({
	options: state.options.map((option) => (option.option))
})

const mapDispatchToProps = (dispatch) => ({
	addOption: (option, id) => dispatch(addOption(option, id))
})

export default connect(mapStateToPops, mapDispatchToProps)(AddOption);