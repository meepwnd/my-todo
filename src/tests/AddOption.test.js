import React from 'react';
import { shallow } from 'enzyme';
import AddOption from '../components/AddOption';

it('renders welcome message', () => {
  const wrapper = shallow(<AddOption />);
  expect(wrapper).toMatchSnapshot();
});