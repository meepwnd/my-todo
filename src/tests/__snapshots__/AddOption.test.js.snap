// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders welcome message 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
